src/log.cpp

```diff
@@ -274,8 +275,39 @@ void Logger::log(LogLevel lev, const std::string &text)
 	const std::string label = getLevelLabel(lev);
 	const std::string timestamp = getTimestamp();
 	std::ostringstream os(std::ios_base::binary);
+
+	static const std::string use_logcolour = g_settings->get("log_colour");
+
+	bool coloured = use_logcolour == "tty" ? isatty(STDOUT_FILENO) : use_logcolour == "yes";
```

Comment:
Well yeah in fact we have no idea here whether we print to stdout, a file, or something else...

Seems that your changes should be done to the `StreamLogOutput` class instead. To detect whether the output is a tty or not add a bool to the constructor and edit the times where the constructor is invoked to use isatty like above.